libass/ass_utils.h

```diff
@@ -159,6 +163,17 @@ static inline uint32_t ass_bswap32(uint32_t x)
 #endif
 }
 
+static inline long ass_lrint(double x)
+{
+#if defined(_MSC_VER) && !defined(_M_ARM64EC) && (defined(_M_X64) || _M_IX86_FP == 2)
```

Comment:
Why the explicit exclusion of arm64 here? Shouldn't the X64/X86 conditions not pass on ARM?

Reply:
When compiling for the ARM64EC ABI the `_M_X64` macro is also defined, as [documented here](https://learn.microsoft.com/en-us/cpp/preprocessor/predefined-macros?view=msvc-170).

The longer explanation is that there are two ABIs on ARM64 Windows. "ARM64" is the normal ABI and it behaves as one would expect. "ARM64EC" is a special ABI that tries to be as X64 compatible as possible to allow linking with emulated X64 binaries.

Reply:
Huh! Bizarre, but I suppose it's harmless. I see that MSVC ends up emitting a call to `_CVTSD2SI32` here, and I wonder how that differs from `lrint`.